### PR TITLE
Fix submission data not uploading to firestore

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
@gino-m  PTAL?

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt [hunk 1]
```diff
@@ -151,7 +151,9 @@ internal constructor(
           responses.forEach { (task, taskData) ->
             taskDataDeltas.add(TaskDataDelta(task.id, task.type, Optional.ofNullable(taskData)))
           }
-          submissionRepository.createOrUpdateSubmission(it, taskDataDeltas.build(), isNew = true)
+          submissionRepository
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt:154]: If the returned subscription is disposed of before the update completes, won't the update process be interrupted? Can we instead introduce an `externalScope` as defined [here](https://medium.com/androiddevelopers/coroutines-patterns-for-work-that-shouldnt-be-cancelled-e26c40f142ad), and call `submissionRepository.createOrUpdateSubmission(..).blockingFoo()` from a coroutine?

It's also not clear why this method returns a Single at all. Could this be simplified?

Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt]: Done

**Comment by shobhitagarwal1612** [comment]
@gino-m PTAL.

**Review by gino-m — CHANGES_REQUESTED** [review]
LG, just one nit. Thanks!

**Review by gino-m — APPROVED** [review]
LGTM!

**Pull request merged**